### PR TITLE
Validate API request query

### LINKED ISSUE #4640: Validate API request query
Current validation middleware only validates request body. Better to enhance it to validate request query as well.

### PR DESCRIPTION
**Pull request by apoorv-mishra**
Closes #4640 

- [x] Fix broken routes
  - [x] `api/events` 
  - [x] `api/documents`
  - [x] `api/attachments`

### CODE REVIEW

File: server/middlewares/validate.ts [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
-export default function validate<T extends z.ZodTypeAny>(schema: T) {
-  return async function validateMiddleware(ctx: APIContext<T>, next: Next) {
+export default function validate<T extends z.ZodType<BaseReq>>(schema: T) {
+  return async function validateMiddleware(
+    ctx: APIContext<BaseReq>,
+    next: Next
+  ) {
     try {
-      ctx.input = schema.parse(ctx.request.body);
+      ctx.input = schema.parse(ctx.request);
```
Comment (apoorv-mishra) [inline server/middlewares/validate.ts:12]: Should probably rename `input` to something more relevant like `validatedReq`.

**Review by apoorv-mishra — COMMENTED** [review]
This, so far is just a preliminary work for initial feeback. The `types` are currently broken due to partial work. In hindsight, it's good because it implies that the failing routes do not comply with the intact _base types_ in place, which guides further dev work to fix these :relieved: 

cc: @tommoor

File: server/middlewares/validate.ts (hunk 1 above)
Comment (tommoor) [reply server/middlewares/validate.ts:12]: I'll happily take the small amount of ambiguity here and keep it as `input`, it's shorter and clear enough for folks working on the codebase every day.

**Comment by tommoor** [comment]
This is great, no notes – feel free to continue 😄

File: server/middlewares/validate.ts [hunk 2]
```diff
@@ -12,7 +12,8 @@ export default function validate<T extends z.ZodType<BaseReq>>(schema: T) {
       ctx.input = schema.parse(ctx.request);
     } catch (err) {
       const { path, message } = err.issues[0];
-      const [prefix = "ValidationError"] = path;
+      const prefix =
+        path.length > 0 ? path[path.length - 1] : "ValidationError";
```
Comment (tommoor) [inline server/middlewares/validate.ts:16]: Isn't `ValidationError` as part of the message a bit redundant here? Perhaps better to just pass `message` in that case…

File: server/routes/api/documents/schema.ts [hunk 3]
```diff
@@ -107,136 +115,154 @@ export const DocumentsInfoSchema = z
...
+  }),
+}).refine((req) => !(req.body.parentDocumentId === req.body.id), {
   message: "infinite loop detected, cannot nest a document inside itself",
 });
 
 export type DocumentsMoveReq = z.infer<typeof DocumentsMoveSchema>;
 
-export const DocumentsArchiveSchema = BaseIdSchema.extend({});
+export const DocumentsArchiveSchema = BaseSchema.extend({
+  body: BaseIdSchema.extend({}),
```
Comment (tommoor) [inline server/routes/api/documents/schema.ts:229]: Unfamiliar, Is it necessary do do the empty extend each time?

**Pull request merged**